tools/mkdmadata.c

```diff
@@ -16,24 +16,38 @@ static void write_dmadata_table(FILE *fout)
 {
     int i;
 
-    for (i = 0; i < g_segmentsCount; i++)
+    for (i = 0; i < g_segmentsCount; i++) {
+        // Don't emit dma entry for segments set with NOLOAD
+        if (g_segments[i].flags & FLAG_NOLOAD) {
+            continue;
+        }
+
         fprintf(fout, "DEFINE_DMA_ENTRY(%s, \"%s\")\n", g_segments[i].name, g_segments[i].name);
+    }
 }
 
 static void write_compress_ranges(FILE *fout)
 {
     int i;
+    int rom_index = 0;
     bool continue_list = false;
     int stride_first = -1;
 
     for (i = 0; i < g_segmentsCount; i++) {
-        if (g_segments[i].compress) {
+        bool compress = g_segments[i].compress;
+
+        // Don't emit dma entry for segments set with NOLOAD
```

Comment:
```suggestion
```
This comment doesn't really apply here. Either remove or perhaps change it, though not sure what to.

Reply:
How about now?